File: yospace/src/main/java/com/bitmovin/player/integration/yospace/DefaultBitmovinYospacePlayerPolicy.java
package com.bitmovin.player.integration.yospace;

public class DefaultBitmovinYospacePlayerPolicy implements BitmovinYospacePlayerPolicy {
    private BitmovinYospacePlayer bitmovinYospacePlayer;

    public DefaultBitmovinYospacePlayerPolicy(BitmovinYospacePlayer player) {
        this.bitmovinYospacePlayer = player;
    }

    @Override
    public boolean canSeek() {
        return !bitmovinYospacePlayer.isAd();
    }

    @Override
    public long canSeekTo(long seekTarget) {
        return seekTarget;
    }

    @Override
    public int canSkip() {
        return 0;
    }

    @Override
    public boolean canPause() {
        return true;
    }

    @Override
    public boolean canMute() {
        return true;
    }
}

File: yospace/src/main/java/com/bitmovin/player/integration/yospace/DefaultBitmovinYospacePlayerPolicy.kt
package com.bitmovin.player.integration.yospace

class DefaultBitmovinYospacePlayerPolicy(private val player: BitmovinYospacePlayer) : BitmovinYospacePlayerPolicy {

    override fun canSeek(): Boolean = !player.isAd()

    override fun canSeekTo(seekTarget: Long): Long = seekTarget

    override fun canSkip(): Int = 0

    override fun canPause(): Boolean = true

    override fun canMute(): Boolean = true
}
